src/librbd/Journal.cc

```diff
@@ -822,51 +822,63 @@ uint64_t Journal<I>::append_io_event(journal::EventEntry &&event_entry,
                                      const AioObjectRequests &requests,
                                      uint64_t offset, size_t length,
                                      bool flush_entry) {
+  Bufferlists *bufferlists = new Bufferlists();
   bufferlist bl;
   ::encode(event_entry, bl);
-  return append_io_events(event_entry.get_event_type(), {bl}, requests, offset,
+  bufferlists->push_back(bl);
+  return append_io_events(event_entry.get_event_type(), bufferlists, requests, offset,
                           length, flush_entry);
 }
 
 template <typename I>
 uint64_t Journal<I>::append_io_events(journal::EventType event_type,
-                                      const Bufferlists &bufferlists,
+                                      Bufferlists *bufferlists,
```

Comment:
I'll need C++14 extensions to use move semantics in the capture list of a lambda function.